cpu/src/main/java/io/smallrye/common/cpu/CacheInfo.java

```diff
@@ -78,82 +78,90 @@ public static int getSmallestInstructionCacheLineSize() {
-                        if (l2 != 0) {
-                            list.add(new CacheLevelInfo(2, CacheType.UNIFIED, l2 / 1024, lineSize));
-                        }
-                        if (l3 != 0) {
-                            list.add(new CacheLevelInfo(3, CacheType.UNIFIED, l3 / 1024, lineSize));
-                        }
-                        if (list.size() > 0) {
-                            return list.toArray(new CacheLevelInfo[list.size()]);
+                    } else if (osArch.contains("mac os x")) {
+                        // cache line size
+                        final int lineSize = safeParseInt(parseProcessOutput("/usr/sbin/sysctl", "-n", "hw.cachelinesize"));
+                        if (lineSize != 0) {
+                            // cache sizes
+                            final int l1d = safeParseInt(parseProcessOutput("/usr/sbin/sysctl", "-n", "hw.l1dcachesize"));
+                            final int l1i = safeParseInt(parseProcessOutput("/usr/sbin/sysctl", "-n", "hw.l1icachesize"));
+                            final int l2 = safeParseInt(parseProcessOutput("/usr/sbin/sysctl", "-n", "hw.l2cachesize"));
+                            final int l3 = safeParseInt(parseProcessOutput("/usr/sbin/sysctl", "-n", "hw.l3cachesize"));
+                            ArrayList<CacheLevelInfo> list = new ArrayList<CacheLevelInfo>();
+                            if (l1d != 0) {
+                                list.add(new CacheLevelInfo(1, CacheType.DATA, l1d / 1024, lineSize));
+                            }
+                            if (l1i != 0) {
+                                list.add(new CacheLevelInfo(1, CacheType.INSTRUCTION, l1i / 1024, lineSize));
+                            }
+                            if (l2 != 0) {
+                                list.add(new CacheLevelInfo(2, CacheType.UNIFIED, l2 / 1024, lineSize));
+                            }
+                            if (l3 != 0) {
+                                list.add(new CacheLevelInfo(3, CacheType.UNIFIED, l3 / 1024, lineSize));
+                            }
+                            if (list.size() > 0) {
+                                return list.toArray(new CacheLevelInfo[list.size()]);
+                            }
                         }
+                    } else if (osArch.contains("windows")) {
+                        // TODO: use the wmic utility to get cache line info
                     }
-                } else if (osArch.contains("windows")) {
-                    // TODO: use the wmic utility to get cache line info
+                } catch (Throwable ignored) {
                 }
-            } catch (Throwable ignored) {
-            }
-            // all has failed
-            return new CacheLevelInfo[0];
-        });
+                // all has failed
+                return new CacheLevelInfo[0];
+            });
+        } else {
+            cacheLevels = new CacheLevelInfo[0];
```

Comment:
Should we return `64` by default?

https://github.com/JCTools/JCTools/blob/a17b56f09cc9ec7d773707578c89f3ca9c11f968/jctools-core/src/main/java/org/jctools/util/PortableJvmInfo.java#L21

Reply:
I was pondering that, but I would like to hear from @dmlloyd who knows how the API was meant to be used

Reply:
Yes let's use 64 by default.

Reply:
I pushed a change. Is this what you had in mind?

Reply:
That works. It would be marginally simpler though if the call to `determineCacheLevel` were put *inside* the existing privileged block and just short-circuits the logic.

Reply:
PR updated